src/main/java/com/google/devtools/build/lib/skyframe/ConfiguredTargetFunction.java

```diff
@@ -264,11 +265,16 @@ public SkyValue compute(SkyKey key, Environment env) throws ConfiguredTargetFunc
         if (rule.getRuleClassObject().supportsPlatforms()) {
           ImmutableSet<Label> requiredToolchains =
               rule.getRuleClassObject().getRequiredToolchains();
+
+          // Collect local (target, rule) constraints for filtering out execution platforms.
+          ImmutableSet<Label> execConstraintLabels =
+              PlatformSemantics.getExecutionPlatformConstraints(rule);
```

Comment:
It is weird that we're calling getExecutionPlatformConstraints on PlatformSemantics here. All the required information is obtained from the rule object so I'd expect this method to live there. OO-ness seems a bit off here.